Mathlib/Algebra/Order/GroupWithZero/Canonical.lean

```diff
@@ -396,14 +413,51 @@ instance instLinearOrderedCommGroupWithZero [LinearOrderedCommGroup α] :
   __ := instLinearOrderedCommMonoidWithZero
   __ := commGroupWithZero
 
-end WithZero
+theorem one_lt_div' [LinearOrderedCommGroupWithZero α] (a : α) {b : α} (hb : b ≠ 0) :
+    1 < a / b ↔ b < a := by
+  rw [← mul_lt_mul_right₀ (zero_lt_iff.mpr hb), one_mul, div_eq_mul_inv, inv_mul_cancel_right₀ hb]
 
-section MultiplicativeNotation
+theorem strictMonoOn_zpow [LinearOrderedCommGroup α] {n : ℤ} (hn : 0 < n) :
+    StrictMonoOn (fun x : (WithZero α) ↦ x ^ n) (Set.Ioi 0) :=
+  fun a ha b _ hab ↦ by
+    have han : a ^ n ≠ 0 := by
+      obtain ⟨x, hx⟩ := ne_zero_iff_exists (x := a).mp (ne_of_gt ha)
+      rw [← hx, ← coe_zpow, ne_zero_iff_exists]
+      use x ^ n
+    simpa only [← one_lt_div' (b ^ n) han, ← div_zpow] using
+      one_lt_zpow (one_lt_div' _ (ne_of_gt ha)|>.mpr hab) hn
 
-/-- Notation for `WithZero (Multiplicative ℕ)` -/
-scoped[Multiplicative] notation "ℕₘ₀" => WithZero (Multiplicative ℕ)
 
-/-- Notation for `WithZero (Multiplicative ℤ)` -/
-scoped[Multiplicative] notation "ℤₘ₀" => WithZero (Multiplicative ℤ)
+theorem zpow_left_injOn {α : Type*} [LinearOrderedCommGroup α] {n : ℤ} (hn : n ≠ 0) :
+    Set.InjOn (fun x : (WithZero α) ↦ x ^ n) (Set.Ioi 0) := by
+  rcases hn.symm.lt_or_lt with h | h
+  · exact (strictMonoOn_zpow h).injOn
+  · refine fun a ha b hb (hab : a ^ n = b ^ n) ↦ (strictMonoOn_zpow (neg_pos.mpr h)).injOn ha hb ?_
+    simp only [zpow_neg, zpow_neg, hab]
 
-end MultiplicativeNotation
+open Set in
+theorem zpow_left_inj {α : Type*} [LinearOrderedCommGroup α] {n : ℤ} {a b : WithZero α}
+    (ha : a ≠ 0) (hb : b ≠ 0) (hn : n ≠ 0) : a ^ n = b ^ n ↔ a = b :=
+  InjOn.eq_iff (zpow_left_injOn hn) (mem_Ioi.mpr (zero_lt_iff.mpr ha))
+    (mem_Ioi.mpr (zero_lt_iff.mpr hb))
+
+section Int
+
+open Int Multiplicative
+
+theorem lt_succ_iff_le  (x : ℤₘ₀) (m : ℤ) :
+    x < ↑(↑(ofAdd (m + 1)) : ℤₘ₀) ↔ x ≤ (↑(ofAdd m) : ℤₘ₀) := by
```

Comment:
This statement looks really convoluted. What about something like
```suggestion
theorem lt_succ_iff_le {x : ℤₘ₀} {m : Multiplicative ℤ} : x < m * ofAdd (1 : ℤ) ↔ x ≤ m := by
```

Reply:
Nice!